File: source_code - v1/FUNC_Plot_model_2D.m
function FUNC_Plot_model_2D(MODEL, Area, mode)
SAT = MODEL.SAT; UE = MODEL.UE;
if(mode=="normal")
    x_area = Area(1); y_area = Area(2);
    % plot the considered area
    plot([0 x_area x_area 0 0],[0 0 y_area y_area 0],'k','markersize',8,'Linewidth',4);
    hold on;
    for m = 1:size(SAT,2)
        plot(SAT(1,m),SAT(2,m),'r-^','markersize',10,'Linewidth',4);
    end
    plot(UE(1,:),UE(2,:),'b^','markersize',8,'Linewidth',1.5);
    grid on
elseif(mode=="cluster")
    x_area = Area(1); y_area = Area(2);
    % plot the considered area
    plot([0 x_area x_area 0 0],[0 0 y_area y_area 0],'k','markersize',8,'Linewidth',4);
    hold on;
    plot(UE(1,:),UE(2,:),'b^','markersize',8,'Linewidth',1.5);
    IdClusUE = MODEL.IdClusUE;
    NumUE_in_SAT = MODEL.NumUE_in_SAT;
    for u = 1:size(UE,2)
        m = IdClusUE(u);
        plot([UE(1,u) SAT(1,m)], [UE(2,u) SAT(2,m)],'k','Linewidth',2);
    end
    for m = 1:size(SAT,2)
        plot(SAT(1,m),SAT(2,m),'r-^','markersize',10,'Linewidth',4);
    end
    grid on
end
end

File: source_code - v1/FUNC_Plot_model_2D.py
import matplotlib.pyplot as plt


def FUNC_Plot_model_2D(MODEL, Area, mode):
    sat = MODEL["SAT"]
    ue = MODEL["UE"]
    x_area = Area[0]
    y_area = Area[1]
    plt.plot([0, x_area, x_area, 0, 0], [0, 0, y_area, y_area, 0], "k", markersize=8, linewidth=4)
    plt.grid(True)

    if mode == "normal":
        for m in range(sat.shape[1]):
            plt.plot(sat[0, m], sat[1, m], "r-^", markersize=10, linewidth=4)
        plt.plot(ue[0, :], ue[1, :], "b^", markersize=8, linewidth=1.5)
    elif mode == "cluster":
        plt.plot(ue[0, :], ue[1, :], "b^", markersize=8, linewidth=1.5)
        id_clus_ue = MODEL["IdClusUE"]
        for u in range(ue.shape[1]):
            m = id_clus_ue[u]
            plt.plot([ue[0, u], sat[0, m - 1]], [ue[1, u], sat[1, m - 1]], "k", linewidth=2)
        for m in range(sat.shape[1]):
            plt.plot(sat[0, m], sat[1, m], "r-^", markersize=10, linewidth=4)
